File: docs/examples/record_api_ts/src/list.ts
import type { Client, ListResponse } from "trailbase";

export const list = async (client: Client): Promise<ListResponse<object>> =>
  await client.records("movies").list({
    pagination: {
      limit: 3,
    },
    order: ["rank"],
    filters: [
      {
        column: "watch_time",
        op: "lessThan",
        value: "120",
      },
      {
        column: "description",
        op: "like",
        value: "%love%",
      },
    ],
  });

File: docs/examples/record_api_dart/lib/src/list.dart
import 'package:trailbase/trailbase.dart';

Future<ListResponse> list(Client client) async =>
    await client.records('movies').list(
      pagination: Pagination(limit: 3),
      order: ['rank'],
      filters: [
        Filter(
          column: 'watch_time',
          op: CompareOp.lessThan,
          value: '120',
        ),
        Filter(
          column: 'description',
          op: CompareOp.like,
          value: '%love%',
        ),
      ],
    );
